pkg/services/publicdashboards/database/database.go

```diff
@@ -37,6 +37,27 @@ func ProvideStore(sqlStore *sqlstore.SQLStore) *PublicDashboardStoreImpl {
 	}
 }
 
+// Gets list of public dashboards by orgId
+func (d *PublicDashboardStoreImpl) ListPublicDashboards(ctx context.Context, orgId int64) ([]PublicDashboardListResponse, error) {
+	resp := make([]PublicDashboardListResponse, 0)
+
+	err := d.sqlStore.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		sess.Table("dashboard_public").
+			Join("LEFT", "dashboard", "dashboard.uid = dashboard_public.dashboard_uid AND dashboard.org_id = dashboard_public.org_id").
+			Cols("dashboard_public.uid", "dashboard_public.access_token", "dashboard_public.dashboard_uid", "dashboard_public.is_enabled", "dashboard.title").
+			Where("dashboard_public.org_id = ?", orgId)
```

Comment:
We are trying not to use joins in order to eventually move to a document store database / pattern (that won't support joins). 

Is there a way you can do this in two requests? (one fetching public dashboards, and one fetching dashboards via search (supports uid array). 

@ryantxu Hm. this is an interesting scenario for the object store interface, list response that includes reference summery (to support list views that contain info on from a reference)

Reply:
This seems expensive and not ideal to do without a join. That is done with a record in the public dashboard table. We don't store the name of the dashboard. Additionally this is going to become more complicated/expensive in the future depending on if public dashboards ends up many:many with dashboards.